src/hip/hip_build_utils.cpp

```diff
@@ -174,20 +171,20 @@ static fs::path HipBuildImpl(boost::optional<TmpDir>& tmp_dir,
 #endif
 
     // hip version
-    params +=
-        std::string(" -DHIP_PACKAGE_VERSION_FLAT=") + std::to_string(HIP_PACKAGE_VERSION_FLAT);
-
-    params += " ";
+    params += " -DHIP_PACKAGE_VERSION_FLAT=" + std::string{HIP_PACKAGE_VERSION_FLAT} + " ";
     auto bin_file = make_object_file_name(tmp_dir.get() / filename);
 
     // compile
     {
-        const std::string redirector = testing_mode ? " 1>/dev/null 2>&1" : "";
-        const std::string cmd        = env + std::string(" ") + MIOPEN_HIP_COMPILER;
-        const std::string args       = params + filename + " -o " + bin_file + redirector;
-        tmp_dir->Execute(cmd, args);
+        std::string args = params + filename + " -o " + bin_file;
+#ifndef _WIN32
+        // Windows uses WIN32 API to execute a subprocess. None command shell is spawned.
+        if(testing_mode)
+            args += " 1>/dev/null 2>&1";
+#endif
+        tmp_dir->Execute(MIOPEN_HIP_COMPILER, args);
         if(!fs::exists(bin_file))
-            MIOPEN_THROW("Failed cmd: '" + cmd + "', args: '" + args + '\'');
+            MIOPEN_THROW("Failed cmd: '" MIOPEN_HIP_COMPILER "', args: '" + args + '\'');
```

Comment:
ref: #2958

Reply:
@junliume @apwojcik Yeah, this is optimization by-product of this PR that leads to an issue. It is similar to the one with `HIP_PACKAGE_VERSION_FLAT` (see https://github.com/ROCm/MIOpen/pull/2905#discussion_r1575449537 above). We should assume that `MIOPEN_HIP_COMPILER` expands to an expression that can be used as an initializer for std:string, but the assumption here is that it expands to a string literal.